pkg/sql/sem/eval/deps.go

```diff
@@ -536,17 +536,14 @@ type ChangefeedState interface {
 // builtin functions to create, configure, and destroy tenants. The methods will
 // return errors when run by any tenant other than the system tenant.
 type TenantOperator interface {
-	// CreateTenantWithID attempts to install a new tenant in the system, with the
-	// provided tenantID. It returns an error if the tenant already exists. The
-	// new tenant is created at the current active version of the cluster
-	// performing the create.
-	CreateTenantWithID(ctx context.Context, tenantID uint64, tenantName roachpb.TenantName) error
-
-	// CreateTenant attempts to install a new tenant in the system and returns the
-	// ID that is assigned to the tenant. It returns an error if another tenant
-	// with `tenantName` already exists. The new tenant is created at the current
-	// active version of the cluster performing the create.
-	CreateTenant(ctx context.Context, tenantName roachpb.TenantName) (roachpb.TenantID, error)
+	// CreateTenant attempts to install a new tenant in the system.
+	// If the provided tenantID is non-zero, it will be used for the new tenant;
+	// an error will be returned if a tenant with the same ID already exists.
+	// Otherwise, a new tenant ID is allocated.
+	// If the provided enantName is non-empty, it will be used for the new tenant.
+	// The new tenant is created at the current active version of the
+	// cluster performing the create.
+	CreateTenant(ctx context.Context, tenantID uint64, tenantName roachpb.TenantName) (roachpb.TenantID, error)
```

Comment:
Not sure how I feel about using 0 as a sentinel value here.  But, I don't feel strongly about it.